src/style-spec/expression/types/resolved_image.js

```diff
@@ -18,8 +18,9 @@ export default class ResolvedImage {
         return this.name;
     }
 
-    static fromString(name: string): ResolvedImage {
-        return new ResolvedImage({name, available: false});
+    static fromString(name: string, availableImages?: ?Array<string>): ResolvedImage {
+        const available = availableImages ? availableImages.indexOf(name) > -1 : false;
+        return new ResolvedImage({name, available});
```

Comment:
Is this change (including `coercion` expression) related to the PR, or an independent one?

Also noting `availableImages.indexOf` can have performance implications — how often is this method called and in what cases?